### PR TITLE
Update omegaconf version to 2.3.0

### PR DESCRIPTION
**Pull request by RyoYang**
Update `omegaconf` version to [2.3.0](https://pypi.org/project/omegaconf/2.3.0/) as omegaconf 2.0.6 has a non-standard dependency specifier PyYAML>=5.1.*. pip 24.1 will enforce this behaviour change.
Discussion can be found at https://github.com/pypa/pip/issues/12063

image

### CODE REVIEW

File: superbench/runner/runner.py [hunk 1]
```diff
@@ -441,11 +441,11 @@ def _run_proc(self, benchmark_name, mode, vars):
             int: Process return code.
         """
         mode.update(vars)
-        if mode.name == 'mpi' and mode.pattern:
+        if mode.name == 'mpi' and 'pattern' in mode:
```
Comment (guoshzhao) [inline superbench/runner/runner.py:444]: mode.name always exists?

**Review by guoshzhao — APPROVED** [review]

File: superbench/runner/runner.py (hunk 1 above)
Comment (RyoYang) [reply superbench/runner/runner.py:444]: Yes,  according to the schema of mode here: https://github.com/microsoft/superbenchmark/blob/main/docs/superbench-config.mdx#:~:text=parallel%3A%20yes-,name,-Mode%20name%20to

**Review by cp5555 — APPROVED** [review]

File: setup.py [hunk 2]
```diff
@@ -198,7 +198,7 @@ def run(self):
                 'pydocstyle>=5.1.1',
                 'pytest-cov>=2.11.1',
                 'pytest-subtests>=0.4.0',
-                'pytest>=6.2.2',
+                'pytest>=6.2.2, <=7.4.4',
```
Comment (cp5555) [inline setup.py:201]: It's strange to set the upper version.

**Review by cp5555 — CHANGES_REQUESTED** [review]
Strange to set the upper version

File: setup.py (hunk 2 above)
Comment (RyoYang) [reply setup.py:201]: Pytest version 8.2.2 introduced an issue (https://github.com/pytest-dev/pytest/issues/12289) that broke Knack (Issue: https://github.com/microsoft/knack/issues/287). As a result, the pipeline job for Python 3.8 fails due to this issue:
image
I have contacted the Knack team to address this issue. Limiting the pytest version to 7.4.4 (align with python3.7) is a temporary workaround to ensure the SB pipeline passes until Knack provides a fix.

**Comment by yukirora** [comment]
Is this change compatible with the old pip like 22.0.1?

**Review by cp5555 — APPROVED** [review]

**Comment by RyoYang** [comment]
> Is this change compatible with the old pip like 22.0.1?
It should be compatible, cpu-unit-test pipeline (Job python3.6) passed all ut and it used a lower version of pip 21.2.4.

**Comment by dong0321** [comment]
image

Seems to be the same issue. I am building release/0.10 with rocm6.0 dockerfile.

**Comment by dong0321** [comment]
@RyoYang could please check if this need to be fixed also in release/0.10

**Comment by dong0321** [comment]
Build successed after change to 2.3.0.

**Pull request merged**